spec/type.dd

```diff
@@ -274,6 +274,33 @@ $(H3 $(LEGACY_LNAME2 Integer Promotions, integer-promotions, Integer Promotions)
     column.
     )
 
+    $(P Integer promotion applies to each operand of a binary expression:)
+
+    $(SPEC_RUNNABLE_EXAMPLE_COMPILE
+    ---
+    byte a;
+    auto b = a + a;
+    static assert(is(typeof(b) == int));
+    // error: can't implicitly convert expression of type int to byte:
+    //byte c = a + a;
+
+    ushort d;
+    // error: can't implicitly convert expression of type int to ushort:
+    //d = d * d;
+    int e = d * d; // OK
+    static assert(is(typeof(int() * d) == int));
+
+    dchar f;
+    static assert(is(typeof(f - f) == uint));
+    ---
+    )
+
+    $(RATIONALE
+    * 32-bit integer operations are often faster than smaller integer types
```

Comment:
Are there even instructions to do sub 32-bit arithmetic operations?

Reply:
Well gcc uses the `al` 8 or `ax` 16-bit registers for `add`:
https://d.godbolt.org/z/77zxcooon

Here's the link I saw before writing that:
https://stackoverflow.com/questions/5069489/performance-of-built-in-types-char-vs-short-vs-int-vs-float-vs-double

Happy to remove this line if you prefer.

Reply:
https://stackoverflow.com/questions/45660139/how-exactly-do-partial-registers-on-haswell-skylake-perform-writing-al-seems-to

Using the partial/sub registers can lead to some confusing performance traps.